### ISSUE TITLE
GetEntity with composite primary key

### ISSUE THREAD

**Issue by jasonlaw** [rivantsov/vita#124]
Hi @rivantsov ,

I have the following entity schema.
~~~C#
[Entity]
interface ICommunity
{
  [PrimaryKey]
   Guid Key {get; set;} 
}

[Entity]
interface IUser
{
  [PrimaryKey]
   Guid Key {get; set;} 
}

[Entity, PrimaryKey("Community, User", Clustered = true)]
interface ICommunityUser
{
  ICommunity Community {get; set;}
  IUser User {get; set;}
}

[Entity, PrimaryKey("CommunityUser, BlockedUser")]
interface IBlocked
{
  ICommunityUser CommunityUser {get; set;}
  IUser BlockedUser {get; set;}
}
~~~

I can use CreatePrimaryKey to get ICommunityUser entity.
~~~C#
var key = session.CreatePrimaryKey<ICommunityUser>( communityKey, userKey );
var communityUser = session.GetEntity<ICommunityUser>(key);
~~~ 

However, I have no idea how to get IBlocked entity with that, I tried below but it failed.
~~~C#
var key = session.CreatePrimaryKey<IBocked>( communityKey, userKey, blockedUserKey );
var blocked = session.GetEntity<IBocked>(key);
~~~ 

Error:
~~~
System.Exception: Invalid number of key values, expected: 2, provided: 3.
   at Vita.Entities.Util.Throw(String message, Object[] args)
   at Vita.Entities.Util.Check(Boolean condition, String message, Object[] args)
   at Vita.Entities.Runtime.EntityKey..ctor(EntityKeyInfo key, Object[] values)
   at Vita.Entities.EntitySessionExtensions.CreatePrimaryKey[TEntity](IEntitySession session, Object[] values)
~~~

May I know how to get IBlocked with GetEntity?

**Comment by rivantsov** [comment]
yah, looks like there's a bug there. Try this version, put it somewhere in your ext class: 

`public static object CreatePrimaryKeyFixed<TEntity>(this IEntitySession session, params object[] values) {
      Util.CheckParam(values, "values");
      var entInfo = session.Context.App.Model.GetEntityInfo(typeof(TEntity), throwIfNotFound: true);
      var pk =  EntityKey.Create(entInfo.PrimaryKey, values);
      return pk;
    }
`

but... I would question the design here. So, for blocked user, its Primary key contains UserId, twice, the same user? You have 16*3 = 48 byte primary key, 2 identical 16 segments
Would it be better just to add IsBlocked flag to link table ICommunityUser ? then, you have to do some extra to get blocked or unblocked users, but still looks like a better solution to me.

**Comment by jasonlaw** [comment]
Hi @rivantsov ,

Thanks for the swift reply! 

As for the design, it is different user in the IBlocked. The idea is a community user can block any other users in the same community. I have tried but failed to meet the requirement by just adding the IsBlocked flag as suggested. May be I am missing something in the design. Anyway, thanks for the suggestion.

Thank you!

**Comment by rivantsov** [comment]
should be fixed now in 3.03
